### DISCUSSION TITLE
Just a question

### DISCUSSION THREAD

**Discussion by ketchupspill** [community/community#21412] [Other Feature Feedback, Questions, & Ideas]
<p>So I was writing a free response practice question for the AP CSA exam, and one of the questions asked, it wanted me to create a class Outfit that can take in a Pants, Top, and Shoes to then print out, in that order, along with a price total(irrelevant rn). However, the answer code was something like</p>
<p>public class Outfit extends ClothingItem{</p>
<p>private Shoes shoes;<br>
private Pants pants;<br>
private Top top;</p>
<p>public Outfit(btw this is a overloaded method right?)(Shoes aShoes, Pants aPants, Top aTop){</p>
<pre><code>shoes = aShoes;
</code></pre>
<p>pants = aPants;<br>
top = aTop;<br>
}<br>
}</p>
<p>so what i want to know is, why was private Shoes shoes, top, pants included, and also why was it written as Shoes aShoes, etc, in the method Outfit instead of just Shoes?</p>
<p>thanks!</p>

**Comment by billythekid** [comment]
<p>Hi <a class="mention" href="/ketchupspill">@ketchupspill</a>, and welcome to the forum.</p>
<p>These are great OOP questions.</p>
<blockquote>
<p>why was private Shoes shoes, top, pants included</p>
</blockquote>
<p>These are the properties of that class, each line states</p>
<ul>
<li>the property’s visibility - private</li>
<li>the property’s type - Shoes/Pants/Top</li>
<li>the property’s name - shoes/pants/top</li>
</ul>
<p>When you make an instance (object) of this class you’ll be accessing those properties in some way.</p>
<blockquote>
<p>why was it written as Shoes aShoes, etc, in the method Outfit instead of just Shoes?</p>
</blockquote>
<p>I’m not sure what exactly you’re asking about why the <code>Shoes aShoes</code> is required so I’m interpreting this in 3 ways:</p>
<ol>
<li>Why is <code>Shoes</code> needed?</li>
</ol>
<p>In strongly-typed languages the method needs to know what type of argument it’s receiving.</p>
<ol start="2">
<li>Why is <code>aShoes</code> needed?</li>
</ol>
<p>We need to let the method know what variable name it can access for the thing(an instance(object) of Shoes) that’s passed in.</p>
<ol start="3">
<li>Why use <code>aShoes</code> and not just <code>Shoes</code> or <code>shoes</code>?</li>
</ol>
<p>Clarity and convention mostly. It shows you want to put a single instance of that class in the constructor. You could use any variable names here but this <code>aThing</code> convention just makes it easy to remember what it was when you come to use it later. You don’t need to ask, what did I call that variable for the Shoes object again? You already know because you stuck to the convention.</p>
<p>Does that help?</p>
<p>Cheers,<br>
Billy :cowboy_hat_face:</p>

**Comment by fire-eggs** [comment]
<p><code>Shoes</code> is a type name and so can’t be used as a variable name. I.e. the compiler would be confused by <code>Shoes Shoes</code>.</p>
<p>You can’t use <code>shoes</code> as the parameter name because of the confusion with the class member.</p>
<p><code>aShoes</code> is not a great choice of name; <code>Shoes someShoes</code> would be clearer to me.</p>
<p>So:</p>
<pre><code class="lang-auto">public Outfit(Shoes Shoes, Pants Pants, Top Top)
</code></pre>
<p>isn’t allowed.</p>
<pre><code class="lang-auto">public Outfit(Shoes shoes, Pants pants, Top top)
{ shoes = shoes; // major confusion!
</code></pre>
<p>The class members (<code>private Shoes shoes</code>) is there to hold ‘state’. That state information would be accessed by other class methods, such as printing the outfit (in the desired order) or printing the price. Following OOP, those functions would not be performed in the Outfit() constructor.</p>
